internal/service/user_test.go

```diff
@@ -0,0 +1,137 @@
+package service
+
+import (
+	"context"
+	"testing"
+
+	"github.com/flexprice/flexprice/internal/domain/user"
+	"github.com/flexprice/flexprice/internal/testutil"
+	"github.com/stretchr/testify/suite"
+)
+
+type UserServiceSuite struct {
+	suite.Suite
+	ctx         context.Context
+	userService *userService
+	userRepo    *testutil.InMemoryUserRepository
+}
+
+func TestUserService(t *testing.T) {
+	suite.Run(t, new(UserServiceSuite))
+}
+
+func (s *UserServiceSuite) SetupTest() {
+	s.ctx = testutil.SetupContext()
+	s.userRepo = testutil.NewInMemoryUserRepository()
+
+	s.userService = &userService{
+		userRepo: s.userRepo,
+	}
+}
+
+func (s *UserServiceSuite) TestCreateUser() {
+	testCases := []struct {
+		name          string
+		email         string
+		setup         func()
+		expectedError bool
+	}{
+		{
+			name:          "successful_creation",
+			email:         "newuser@example.com",
+			expectedError: false,
+		},
+		{
+			name:  "duplicate_email",
+			email: "existing@example.com",
+			setup: func() {
+				// Pre-insert a user to simulate a duplicate
+				_ = s.userRepo.Create(s.ctx, &user.User{
+					ID:    "user-1",
+					Email: "existing@example.com",
+				})
+			},
+			expectedError: true,
+		},
+		{
+			name:          "empty_email",
+			email:         "",
+			expectedError: true, // Assumes userService should not allow empty emails
+		},
+		{
+			name:          "invalid_email_format",
+			email:         "not-an-email",
+			expectedError: true, // Assumes userService should validate email format
+		},
+		{
+			name:          "very_long_email",
+			email:         "this_is_a_very_long_email_address_that_might_break_some_limitations@somewhere.com",
+			expectedError: false, // If you decide to allow any length, keep false; else true.
+		},
```

Comment:
you can remove the test cases for which the service code handling is not there yet.